### ISSUE TITLE
Error with Screen Orientation when building on IOS

### ISSUE THREAD

**Issue by CurtisConway** [driftyco/ionic-native#472]
As mentioned on the ionic forums Screen Orientation is behaving perfectly on Android, no complaints there but when I try to build the app on IOS I get a console error beyond my comprehension.

[Error] Unhandled Promise rejection: (7)
consoleError (zone.js:461)
loop1 (zone.js:490)
drainMicroTaskQueue (zone.js:494)
g (es6-shim.min.js:11:1948)
(anonymous function) (es6-shim.min.js:11:1878)
promiseReactionJob
[Error] Error: Uncaught (in promise): EXCEPTION: Error in build/pages/live/live.html:3:22
ORIGINAL EXCEPTION: TypeError: undefined is not an object (evaluating 'util_1.get(window, pluginObj.pluginRef)[methodName].apply')
ORIGINAL STACKTRACE:
callCordovaPlugin@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:93433:63
file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:93573:37
value@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:93636:66
appSetup@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:944:57
ngOnInit@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:528:35
detectChangesInternal
detectChanges@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:32570:35
detectChanges@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:32675:48
detectViewChildrenChanges@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:32596:32
detectChangesInternal
detectChanges@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:32570:35
detectChanges@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:32675:48
detectChanges@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:33027:78
fireWillEnter@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:63546:35
_postRender@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:61880:43
file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:61829:34
file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:62219:21
loaded@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:63523:13
file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:62202:24
invoke@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/zone.js:323:34
onInvoke@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:38069:47
invoke@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/zone.js:322:43
run@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/zone.js:216:50
file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/zone.js:571:61
invokeTask@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/zone.js:356:43
onInvokeTask@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/app.bundle.js:38060:51
invokeTask@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/zone.js:355:55
runTask@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/zone.js:256:58
drainMicroTaskQueue@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/zone.js:474:43
g@file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/es6-shim.min.js:11:1947
file:///var/containers/Bundle/Application/1A3BDE98-2910-4953-A6F3-6E6AC06293DC/Drumeo%20Live.app/www/build/js/es6-shim.min.js:11:1877
promiseReactionJob@[native code]
ERROR CONTEXT:
[object Object] — zone.js:537
consoleError (zone.js:463)
loop1 (zone.js:490)
drainMicroTaskQueue (zone.js:494)
g (es6-shim.min.js:11:1948)
(anonymous function) (es6-shim.min.js:11:1878)
promiseReactionJob

**Comment by ihadeed** [comment]
Hey @CurtisConway 

Can you provide the following:
- Code snippet
- Platform information: what device are you running on? 
- ionic-native version

**Comment by CurtisConway** [comment]
Platform: IPhone 5 running IOS 9.3.4
Ionic Native Version. 1.3.17

My two functions utilising the plugin:

```
appSetup(v:string) {
        this.videoElement = <HTMLVideoElement> document.getElementById(v);
        this.videoElement.addEventListener("timeupdate", this.updateTime);
        
        // Lock Screen Orientation in Portrait
        ScreenOrientation.lockOrientation('portrait');
    };
```

```
fullScreen() {
        let mainBody = <HTMLElement>document.querySelector("ion-page");

        if(this.isFullScreen == false) {
            this.isFullScreen = true;
            console.log("Going Fullscreen");


            mainBody.classList.toggle("is-fullscreen");
            ScreenOrientation.unlockOrientation();

        }
        else {
            this.isFullScreen = false;
            console.log("Exiting Fullscreen");

            mainBody.classList.toggle("is-fullscreen");
            ScreenOrientation.lockOrientation('portrait');
        }
    };
```

**Comment by CurtisConway** [comment]
Just a quick update. When I first posted this issue I was still running Beta10, I just spent the morning updating to Beta11 aswell as updating to Ionic Native version 1.3.20. The app does work when I initially run it through Xcode now but I try to refresh through the console or restart the app it breaks again, same error. It's also worth noting that the plugin doesn't even work when the app does load up.

**Comment by ihadeed** [comment]
You should wrap the method call with `this.platform.ready().then(() => { ... });`

**Comment by ghenry22** [comment]
If you are trying to restrict the orientation to portrait, as it looks from the snippet, have you tried just setting orientation in the config.xml?  Alternatively for iOS open up in Xcode and choose the allowed orientations, once done it will suck unless you remove/add platform.

**Comment by ihadeed** [comment]
This issue should be resolved by using Ionic Native 3.1.x and latest version of the plugin.
